exporter/exporterparser/jaeger.go

```
@@ -0,0 +1,87 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporterparser
+
+import (
+	"context"
+	"log"
+
+	commonpb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1"
+	"github.com/census-instrumentation/opencensus-service/exporter"
+	"go.opencensus.io/exporter/jaeger"
+	"go.opencensus.io/trace"
+)
+
+// Slight modified version of go/src/go.opencensus.io/exporter/jaeger/jaeger.go
+type jaegerConfig struct {
+	Endpoint          string `yaml:"endpoint,omitempty"`
+	CollectorEndpoint string `yaml:"collector_endpoint,omitempty"`
+	AgentEndpoint     string `yaml:"agent_endpoint,omitempty"`
+	Username          string `yaml:"username,omitempty"`
+	Password          string `yaml:"password,omitempty"`
+	ServiceName       string `yaml:"service_name,omitempty"`
+}
+
+type jaegerExporter struct {
+	exporter *jaeger.Exporter
+}
+
+// JaegerExportersFromYAML reads any configured jaeger exporters from a yaml config
+func JaegerExportersFromYAML(config []byte) (tes []exporter.TraceExporter, doneFns []func() error, err error) {
+	var cfg struct {
+		Exporters *struct {
+			Jaeger *jaegerConfig `yaml:"jaeger"`
+		} `yaml:"exporters"`
+	}
+	if err := yamlUnmarshal(config, &cfg); err != nil {
+		return nil, nil, err
+	}
+	if cfg.Exporters == nil {
+		return nil, nil, nil
+	}
+	jc := cfg.Exporters.Jaeger
+	if jc == nil {
+		return nil, nil, nil
+	}
+
+	// jaeger.NewExporter performs configurqtion validation
+	je, err := jaeger.NewExporter(jaeger.Options{
+		Endpoint:          jc.Endpoint,
+		CollectorEndpoint: jc.CollectorEndpoint,
+		AgentEndpoint:     jc.AgentEndpoint,
+		Username:          jc.Username,
+		Password:          jc.Password,
+		Process: jaeger.Process{
+			ServiceName: jc.ServiceName,
+		},
+	})
+	if err != nil {
+		log.Fatalf("Cannot configure Jaeger exporter: %v", err)
```

Please return this error instead of `log.Fatalf` so `return nil, nil, err`